process/block/preprocess/transactions_test.go

```diff
@@ -2126,3 +2127,71 @@ func createMockTransactions(numTxs int, sndShId byte, rcvShId byte, startNonce u
 
 	return txs
 }
+
+func TestTransactions_getIndexesOfLastTxProcessed(t *testing.T) {
+	t.Parallel()
+
+	t.Run("calculating hash error should not get indexes", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultTransactionsProcessorArgs()
+		args.Marshalizer = &testscommon.MarshalizerMock{
+			Fail: true,
+		}
+		txs, _ := NewTransactionPreprocessor(args)
+
+		miniBlock := &block.MiniBlock{}
+		pmbt := &processedMb.ProcessedMiniBlockTracker{}
+		headerHandler := &block.Header{}
+
+		pi, err := txs.getIndexesOfLastTxProcessed(miniBlock, pmbt, headerHandler)
+		assert.Nil(t, pi)
+		assert.Equal(t, testscommon.ErrMockMarshalizer, err)
+	})
+
+	t.Run("missing mini block header should not get indexes", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultTransactionsProcessorArgs()
+		args.Marshalizer = &testscommon.MarshalizerMock{
+			Fail: false,
+		}
+		txs, _ := NewTransactionPreprocessor(args)
+
+		miniBlock := &block.MiniBlock{}
+		pmbt := &processedMb.ProcessedMiniBlockTracker{}
+		headerHandler := &block.Header{}
+
+		pi, err := txs.getIndexesOfLastTxProcessed(miniBlock, pmbt, headerHandler)
+		assert.Nil(t, pi)
+		assert.Equal(t, process.ErrMissingMiniBlockHeader, err)
+	})
+
+	t.Run("should get indexes", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultTransactionsProcessorArgs()
+		args.Marshalizer = &testscommon.MarshalizerMock{
+			Fail: false,
+		}
+		txs, _ := NewTransactionPreprocessor(args)
+
+		miniBlock := &block.MiniBlock{}
+		miniBlockHash, _ := core.CalculateHash(txs.marshalizer, txs.hasher, miniBlock)
+		mbh := block.MiniBlockHeader{
+			Hash:    miniBlockHash,
+			TxCount: 6,
+		}
+		_ = mbh.SetIndexOfFirstTxProcessed(2)
+		_ = mbh.SetIndexOfLastTxProcessed(4)
+		pmbt := &processedMb.ProcessedMiniBlockTracker{}
+		headerHandler := &block.Header{
+			MiniBlockHeaders: []block.MiniBlockHeader{mbh},
+		}
+
+		pi, err := txs.getIndexesOfLastTxProcessed(miniBlock, pmbt, headerHandler)
+		assert.Nil(t, err)
+		assert.Equal(t, int32(-1), pi.indexOfLastTxProcessedByItself)
```

Comment:
please also rename `indexOfLastTxProcessedByItself` to `indexOfLastProcessedTx` or to be consistent with mb 
`indexOfLastTxProcessed`

Reply:
Done